### PR TITLE
Create the Rockabox date time component

### PR DESCRIPTION
**Pull request by nwhite89**
Creates the Rockabox date time component ``<rb-date-time>`` allowing to specify:
- No time ``show-time="false"``
- Understated time ``understate-time="true"``

TP: https://rockabox.tpondemand.com/entity/8270

### CODE REVIEW

File: src/rb-date-time/rb-date-time-directive.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @restrict E
+     *
+     * @description
+     *
+     * @usage
+     * <hljs lang="html">
+     *    <rb-date-time dateTime="1970-01-01T00:00"></rb-date-time>
+     *    <rb-date-time dateTime="1970-01-01T00:00" understateTime="true"></rb-date-time>
+     *    <rb-date-time dateTime="1970-01-01T00:00" understateTime="true" show-time="false"></rb-date-time>
```
Comment (chrisvxd) [inline src/rb-date-time/rb-date-time-directive.js]: Why the mixture of camel-case and hyphenated?

File: src/rb-date-time/rb-date-time.tpl.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<time
+    class="DateTime"
+    ng-class="understateTime ? 'DateTime--understateTime' : ''"
```
Comment (chrisvxd) [inline src/rb-date-time/rb-date-time.tpl.html]: Wouldn't this be tidier using an object for the expression? `{DateTime--understateTime: understateTime}`

File: src/rb-date-time/rb-date-time-directive.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        return {
+            scope: {
+                dateTime: '@',
+                understateTime: '@',
+                showTime: '@'
+            },
+            restrict: 'E',
+            replace: true,
+            transclude: true,
```
Comment (chrisvxd) [inline src/rb-date-time/rb-date-time-directive.js]: Transclusion isn't needed here

Comment (nwhite89) [inline src/rb-date-time/rb-date-time-directive.js]: true tbh bit of copy pasta lol

File: src/rb-date-time/rb-date-time.tpl.html (hunk 2 above)
Comment (nwhite89) [inline src/rb-date-time/rb-date-time.tpl.html]: that expression wouldn't work :confused: ?

File: src/rb-date-time/rb-date-time-directive.js (hunk 1 above)
Comment (nwhite89) [inline src/rb-date-time/rb-date-time-directive.js]: shouldn't be will change

File: src/rb-date-time/rb-date-time.tpl.html (hunk 2 above)
Comment (chrisvxd) [inline src/rb-date-time/rb-date-time.tpl.html]: Angular processes objects like that for ng-class...

`2. If the expression evaluates to an object, then for each key-value pair of the object with a truthy value the corresponding key is used as a class name.`

https://docs.angularjs.org/api/ng/directive/ngClass

Comment (nwhite89) [inline src/rb-date-time/rb-date-time.tpl.html]: hmm ok will give it a go

File: src/rb-date-time/rb-date-time-directive.js (hunk 3 above)
Comment (nwhite89) [inline src/rb-date-time/rb-date-time-directive.js]: good to note it's in the generator currently we may want to remove that for now? @chrisvxd

**Pull request merged**